robot/impl/local_robot.go

```diff
@@ -45,7 +45,7 @@ type localRobot struct {
 	// statusLock guards calls to the Status method.
 	statusLock    sync.Mutex
 	manager       *resourceManager
-	mostRecentCfg config.Config
+	mostRecentCfg atomic.Value // config.Config
```

Comment:
An atomic value (pointer under the hood) is what I would use here for a different language without hesitation. Or for Go if the type was genericized. I assume it's not only for backwards compatibility.

I think* I still prefer this compared to a very* special mutex just for this variable. With a weird access pattern. Weird in the sense that it's the only member variable of `localRobot` that needs this help.

Reply:
Agree that atomic value is the way to go here (mutex to protect a single field can be an odd pattern).